components/suggestion/generic_channel_provider.tsx

```diff
@@ -1,11 +1,26 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
 
+import { Channel } from '@mattermost/types/channels';
+import { ServerError } from '@mattermost/types/errors';
+import { ActionFunc, ActionResult } from 'mattermost-redux/types/actions';
+
 import React from 'react';
 
 import Provider from './provider';
 import Suggestion from './suggestion.jsx';
 
+export type Results = {
+    matchedPretext: string;
+    terms: string[];
+    items: Channel[];
+    component: React.ElementType;
+}
+
+type ResultsCallback = (results: Results) => void;
+
+type ChannelSearchFunc = (term: string, success: (channels: Channel[]) => void, error: (err: ServerError) => void) => ActionFunc
```

Comment:
```suggestion
type ChannelSearchFunc = (term: string, success: (channels: Channel[]) => void, error?: (err: ServerError) => void) => (ActionResult | Promise<ActionResult | ActionResult[]>);
```

Reply:
Here, making `error` optional in ChannelSearchFunc gives an error in `action_menu.tsx`.

> Types of parameters 'error' and 'error' are incompatible.
Type '((err: ServerError) => void) | undefined' is not assignable to type '(err: ServerError) => void'.
Type 'undefined' is not assignable to type '(err: ServerError) => void'.

And if I don't make `error` optional to align with `action_menu.tsx`, i.e.
```suggestion
type ChannelSearchFunc = ((term: string, success: (channels: Channel[]) => void, error: (err: ServerError) => void) => (ActionResult | Promise<ActionResult | ActionResult[]>);
```

This error shows up.


![image](https://user-images.githubusercontent.com/56581495/215243369-14179068-32c5-4e02-bc6f-c86b02271730.png)

Reply:
>Types of parameters 'error' and 'error' are incompatible.
Type '((err: ServerError) => void) | undefined' is not assignable to type '(err: ServerError) => void'.
Type 'undefined' is not assignable to type '(err: ServerError) => void'.

The reason you are getting the error is that you make it `optional` in one place and not in other places. Therefore you need to align error definition everywhere: `dialog_element`, `action_menu`, `generic_channel_provider` and `actions/channel_actions`. From the error message, try to understand where you are missing `error` definition

Reply:
Oh. Got it👍🏻